airbyte-cdk/python/connector_builder/message_grouper.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import logging
+from copy import deepcopy
+from json import JSONDecodeError
+from typing import Any, Iterable, Iterator, Mapping, Optional, Union
+from urllib.parse import parse_qs, urlparse
+
+from airbyte_cdk.models import AirbyteLogMessage, AirbyteMessage, Type
+from airbyte_cdk.sources.declarative.declarative_source import DeclarativeSource
+from airbyte_cdk.utils.schema_inferrer import SchemaInferrer
+from airbyte_protocol.models.airbyte_protocol import ConfiguredAirbyteCatalog
+from connector_builder.models import HttpRequest, HttpResponse, StreamRead, StreamReadPages, StreamReadSlices
+
+
+class MessageGrouper:
+    logger = logging.getLogger("airbyte.connector-builder")
+
+    def __init__(self, max_pages_per_slice: int, max_slices: int, max_record_limit: int = 1000):
```

Comment:
nit (also I know this may just be copy/pasted): Would it be preferable to enforce the maximums in this class, instead of allowing them to be fully configurable?

Reply:
the main reason why I didn't enforce maximums here is that we'll need to update the value in two places if we decide to increase it. I think the risk of not enforcing a limit here is negligible because we control the caller